src/sage/matroids/oriented_matroids/oriented_matroid.py

```diff
@@ -0,0 +1,1038 @@
+                # If an edge exists in both sets, then this is a false cycle.
+                # This implies we have ee^-1 which is why it's false.
+                # So we only add the true ones.
+                if not p.intersection(n):
+                    data.append([p, n])
+        elif isinstance(data, Matrix):
+            if key != 'chirotope' and key is not None:
+                raise ValueError(
+                    'matrices are currently only implemented using chirotope axioms')
+            key = 'chirotope'
+
+        if key not in OrientedMatroid.keys:
+            raise ValueError("invalid type key")
+
+        # In the following cases, deep_tupler is used since we are using
+        # UniqueRepresentation Which doesn't allow us to have non-hashable things.
+        if key == "covector":
+            from sage.matroids.oriented_matroids.covector_oriented_matroid \
+                import CovectorOrientedMatroid
+            data = deep_tupler(data)
+            if groundset is not None:
+                groundset = deep_tupler(groundset)
+            OM = CovectorOrientedMatroid(data, groundset=groundset)
+        elif key == "circuit":
+            from sage.matroids.oriented_matroids.circuit_oriented_matroid \
+                import CircuitOrientedMatroid
+            data = deep_tupler(data)
+            if groundset is not None:
+                groundset = deep_tupler(groundset)
+            OM = CircuitOrientedMatroid(data, groundset=groundset)
+        elif key == "vector":
+            from sage.matroids.oriented_matroids.vector_oriented_matroid \
+                import VectorOrientedMatroid
+            data = deep_tupler(data)
+            if groundset is not None:
+                groundset = deep_tupler(groundset)
+            OM = VectorOrientedMatroid(data, groundset=groundset)
+        elif key == "real_hyperplane_arrangement":
+            from sage.matroids.oriented_matroids.real_hyperplane_arrangement_oriented_matroid \
+                import RealHyperplaneArrangementOrientedMatroid
+            A = copy.copy(data)
+            if groundset is None:
+                groundset = deep_tupler(A.hyperplanes())
+            else:
+                groundset = deep_tupler(groundset)
+            OM = RealHyperplaneArrangementOrientedMatroid(A, groundset=groundset)
+
+        if OM is None:
+            raise NotImplementedError(
+                f"oriented matroid of type {key} is not implemented")
+
+        return OM
+
+    def __eq__(self, other):
+        """
+        Return whether two oriented matroids are equal.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: C = [[1, 0, -1], [-1, 0, 1],[0, 0, 0]]
+            sage: M = OrientedMatroid(C, key='covector')
+            sage: M == M
+            True
+        """
+        if isinstance(other, type(self)):
+            return self.elements() == other.elements()
```

Comment:
Could you end up with two oriented matroids that have equal elements but unequal groundsets? I am asking because of the hashing uses both.